### ISSUE TITLE
Add display options for identity name/description

### ISSUE THREAD

**Issue by nomis** [thundernest/k-9#6934]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

6.704

### Problem you are trying to solve

It is confusing when the identity description appears instead of your own name, without any obvious reason why.

It is useful to know when the sender doesn't have the correct name to put in the header, but it is not possible to see that without looking at the headers because email addresses matching the identity are always displayed with the identity description/name.

It's possible to show the original name from the email header for addresses other than your own by disabling "Show contact names".

### Suggested solution

Add a "Show identities as" display option with the following choices:
* Identity description [default; current behaviour]
* Identity name
* Original name/email address

The "identity name" is the current behaviour when there is no identity description.

Actual display of the "original name/email address" would be further determined by the "Show correspondent names" option.

**Comment by gschintgen** [comment]
I fully agree that this is a necessary feature. I often get emails related to one of my kids (e.g. sports club or similar) and the only information which kid is actually meant is often only in the "To:" header, e.g. "Child name <my.address@mydomain.lu>". It took me some time to even notice that K9 does this unwanted modification of the recipient address and that the mails do in fact include this crucial information.

Arguably this is also a sender-side issue, but I still have to deal with it anyway. IMHO any email client should present the To: and From: exactly as-is (possibly with an opt-in choice to have them "beautified") 

Hiding the full To: / From: is also a security issue when considering all the scam and phishing mails that find their way to my inbox.

I'm looking way more often at the full headers than I should have to.
